dspace/modules/xmlui/src/main/java/org/dspace/app/xmlui/aspect/discovery/Navigation.java

```diff
@@ -223,5 +226,21 @@ public void addPageMeta(PageMeta pageMeta) throws SAXException,
             }
         }
 
+        // Add recently integrated journals
+        if (JournalUtils.recentlyIntegratedJournals.size() > 0) {
+            int count = 0;
+            for (DryadJournalConcept journalConcept : JournalUtils.recentlyIntegratedJournals) {
+                // we only have room for 4 journals:
+                if (count++ > 3) break;
+
+                String journalName = journalConcept.getSearchableName();
+                String journalString = "@" + URLEncoder.encode(journalName, "UTF-8") + "@" +
```

Comment:
Does this work if the journal name includes `#` or `$`?

Reply:
It doesn't work perfectly, but it's ok.

Reply:
URLEncoder would percent-encode any special characters in the name, but yeah, if the full name at the end has a `$` in it, it'd be a problem.